File: src/main/java/gov/cdc/nccdphp/esurveillance/data/DataLoader.java
package gov.cdc.nccdphp.esurveillance.data;

import gov.cdc.nccdphp.esurveillance.csvDefinition.model.ValueSet;
import gov.cdc.nccdphp.esurveillance.csvDefinition.repository.ValueSetMongoRepo;
import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import java.io.InputStream;
import java.util.Collections;
import java.util.Scanner;

/**
 * @Created - 7/23/18
 * @Author Marcelo Caldas mcq1@cdc.gov
 */

@Component
public class DataLoader {
    @Autowired
    ValueSetMongoRepo valueSetRepo;

    private Log log = LogFactory.getLog(DataLoader.class);


    public void loadDataSets(String filename) {
        InputStream file = getClass().getClassLoader().getResourceAsStream(filename);
        try (Scanner scanner = new Scanner(file)) {
            scanner.nextLine(); //skip first line
            //Create MMG ->
            while (scanner.hasNextLine()) {
                String line = scanner.nextLine();
                String[] values = line.split(",(?=([^\"]*\"[^\"]*\")*[^\"]*$)");



                ValueSet valueSet = new ValueSet(values[0], Collections.emptyList());
                for (int i = 1; i < values.length; i++) { //Add all possible value Sets:
                    String[] vs_code = values[i].split("\\^");
                    valueSet.addChoice(vs_code[0], vs_code[1]);
                }
                valueSetRepo.save(valueSet);
            }
        }
    }

    private Double getDoubleValue(String value) {
        if (value == null || value.trim().length() == 0) {
            return 0.0;
        } else {
            return new Double(value);
        }
    }
}

File: src/main/java/gov/cdc/nccdphp/esurveillance/data/DataLoader.kt
package gov.cdc.nccdphp.esurveillance.data

import com.google.gson.Gson
import gov.cdc.nccdphp.esurveillance.csvDefinition.model.CSVDefinition
import gov.cdc.nccdphp.esurveillance.csvDefinition.model.ValueSet
import gov.cdc.nccdphp.esurveillance.csvDefinition.repository.ValueSetMongoRepo
import gov.cdc.nccdphp.esurveillance.csvDefinition.service.CSVDefinitionService
import org.apache.commons.logging.LogFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component
import java.util.*

/**
 * @Created - 7/23/18
 * @Author Marcelo Caldas mcq1@cdc.gov
 */

@Component
class DataLoader(val valueSetRepo: ValueSetMongoRepo, val fileDefService: CSVDefinitionService) {

    private val log = LogFactory.getLog(DataLoader::class.java)

    fun loadDataSets(content: String) {
        //InputStream file = getClass().getClassLoader().getResourceAsStream(filename);
        Scanner(content).use { scanner ->
            scanner.nextLine() //skip first line
            //Create MMG ->
            while (scanner.hasNextLine()) {
                val line = scanner.nextLine()
                val values = line.split(",(?=([^\"]*\"[^\"]*\")*[^\"]*$)".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()


                val valueSet = ValueSet(values[0])
                for (i in 1 until values.size) { //Add all possible value Sets:
                    val vsCode = values[i].split("\\^".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
                    valueSet.addChoice(vsCode[0], vsCode[1])
                }
                valueSetRepo.save(valueSet)
            }
        }
    }

    fun loadDefinition(content: String) {
        val gson = Gson()
        val newDef =gson.fromJson(content, CSVDefinition::class.java)
        fileDefService.save(newDef)

    }
}
